src/behavior.ts

```diff
@@ -43,94 +39,182 @@ const SpaceCode = "Space";
-        clearCatcher.on("keydown", (e:KeyboardEvent) => {
-            if (e.code !== EnterCode && e.code !== SpaceCode) {
-                return;
-            }
-            this.clear(selectionHandler, onSelect);
+    }
+
+    private setSelectedDataPoints(dataPoints: SunburstDataPoint[]): void{
+        const selectedIds: powerbi.extensibility.ISelectionId[] = this.selectionManager.getSelectionIds();
+        dataPoints.forEach((dp: SunburstDataPoint) => {
+            dp.selected = selectedIds.some((id: ISelectionId) => id.includes(dp.identity));
         });
     }
 
-    public renderSelection(hasSelection: boolean): void {
-        const {
-            selection,
-            interactivityService,
-        } = this.options;
+    public renderSelection(): void {
+        const selection = this.options.selection;
+        const hasSelection: boolean = this.selectionManager.hasSelection();
 
-        const hasHighlights: boolean = interactivityService.hasSelection();
+        this.setSelectedDataPoints(this.options.dataPoints);
 
         selection.style("opacity", (dataPoint: HierarchyRectangularNode<SunburstDataPoint>) => {
-            const { selected, highlight } = dataPoint.data;
-            return getFillOpacity(
-                selected,
-                highlight,
-                !highlight && hasSelection,
-                !selected && hasHighlights
-            );
+            return getFillOpacity(dataPoint.data.selected, hasSelection);
         });
 
         selection.attr("aria-selected", (dataPoint: HierarchyRectangularNode<SunburstDataPoint>) => {
             return dataPoint.data.selected;
         });
+
+        if (this.options.onSelect){
+            const canDisplayCategory: boolean = this.selectionManager.getSelectionIds().length === 1;
+            const label: SunburstLabel = this.createCategoryLabel(canDisplayCategory);
+
+            this.options.onSelect(label, hasSelection, canDisplayCategory);
+        }
+    }
+
+    private createCategoryLabel(canDisplayCategory: boolean): SunburstLabel {
+        if (canDisplayCategory){
+            const selectedId = <ISelectionId>this.selectionManager.getSelectionIds()[0];
+            const selectedDataPoint: SunburstDataPoint = this.options.dataPoints.find((el: SunburstDataPoint) => el.identity.equals(selectedId));
+            const label: SunburstLabel = {
+                text: selectedDataPoint.tooltipInfo[0].displayName,
+                total: selectedDataPoint.total,
+                color: selectedDataPoint.color
+            };
+            return label;
+        }
+        else {
+            const total: number = this.calculateTotalForLabel(this.options.dataPointsTree, 0);
+            const label: SunburstLabel = {
+                text: "",
+                total: total,
+                color: "black"
+            };
+            return label;
+        }
+    }
+
+    private calculateTotalForLabel(dataPoint: SunburstDataPoint, total: number): number {
```

Comment:
add test coverage for this function

Reply:
added 652686fbd36900c8463915a2410315405443a601